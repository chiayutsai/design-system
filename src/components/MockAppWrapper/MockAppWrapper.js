import history from 'history'
import PropTypes from 'prop-types'
import App from 'components/App'
import configureStore from 'store/configureStore'
import { injectReducer } from 'store/makeRootReducer'

const initialState = {}

export const store = configureStore(initialState, { history })

const MockAppWrapper = ({ children }) => (
  <App
    context={{
      pathname: '',
      store,
      injectReducer,
    }}
    insertCss={() => {}}>
    {children}
  </App>
)

MockAppWrapper.propTypes = {
  children: PropTypes.node.isRequired,
}

export default MockAppWrapper
