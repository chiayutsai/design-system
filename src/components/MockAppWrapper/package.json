{
  "name": "MockAppWrapper",
  "version": "0.0.0",
  "private": true,
  "main": "./MockAppWrapper.js"
}
