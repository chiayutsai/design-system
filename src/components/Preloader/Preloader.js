import classnames from 'classnames'
import useStyles from 'isomorphic-style-loader/useStyles'
import PropTypes from 'prop-types'
import styles from './Preloader.scss'

const Preloader = ({ className, isMini }) => {
  useStyles(styles)
  return (
    <div
      className={classnames(styles.preloader, className, {
        [styles['preloader-mini']]: isMini,
      })}>
      <div className={styles.preloader_spiner}>
        <div className={styles.preloader_circles}>
          <div className={styles['preloader_circle-1']} />
          <div className={styles['preloader_circle-2']} />
        </div>
      </div>
    </div>
  )
}

Preloader.propTypes = { className: PropTypes.string, isMini: PropTypes.bool }
Preloader.defaultProps = { className: '', isMini: false }

export default Preloader
