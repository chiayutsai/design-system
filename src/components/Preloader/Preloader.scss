$preloader-color: #0093FC;

@keyframes container-rotate {
  to {
    transform: rotate(360deg)
  }
}

@keyframes fill-unfill-rotate {
  12.5% {
    transform: rotate(135deg)
  }

  25% {
    transform: rotate(270deg)
  }

  37.5% {
    transform: rotate(405deg)
  }

  50% {
    transform: rotate(540deg)
  }

  62.5% {
    transform: rotate(675deg)
  }

  75% {
    transform: rotate(810deg)
  }

  87.5% {
    transform: rotate(945deg)
  }

  to {
    transform: rotate(1080deg)
  }
}

@keyframes left-spin {
  from {
    transform: rotate(130deg)
  }

  50% {
    transform: rotate(-5deg)
  }

  to {
    transform: rotate(130deg)
  }
}

@keyframes right-spin {
  from {
    transform: rotate(-130deg)
  }

  50% {
    transform: rotate(5deg)
  }

  to {
    transform: rotate(-130deg)
  }
}

.preloader {
  display: flex;
  padding: 3rem;
  justify-content: center;
  border-color: $preloader-color;
}

.preloader_spiner {
  display: inline-flex;
  position: relative;
  width: 65px;
  height: 65px;
  align-self: center;
  font-size: 0;
  border-color: inherit;
  animation: container-rotate 1568ms linear infinite;
}

.preloader_circles {
  position: absolute;
  width: 100%;
  height: 100%;
  border-color: inherit;
  opacity: 1;
  animation: fill-unfill-rotate 5332ms cubic-bezier(0.4, 0, 0.2, 1) infinite both;
}

.preloader_circle-1,
.preloader_circle-2 {
  display: inline-block;
  position: relative;
  width: 50%;
  height: 100%;
  overflow: hidden;
  border-color: inherit;

  &::before {
    content: '';
    width: 200%;
    height: 100%;
    border-width: 4px;
    border-style: solid;
    border-color: inherit;
    border-bottom-color: transparent !important;
    border-radius: 50%;
    animation: none;
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    box-sizing: border-box
  }
}

.preloader_circle-1::before {
  left: 0;
  border-right-color: transparent !important;
  transform: rotate(129deg);
  animation: left-spin 1333ms cubic-bezier(0.4, 0, 0.2, 1) infinite both
}

.preloader_circle-2::before {
  left: -100%;
  border-left-color: transparent !important;
  transform: rotate(-129deg);
  animation: right-spin 1333ms cubic-bezier(0.4, 0, 0.2, 1) infinite both
}

.preloader-mini {
  padding: 1rem;

  .preloader_spiner {
    width: 20px;
    height: 20px;
  }

  .preloader_circle-1::before,
  .preloader_circle-2::before {
    border-width: 2px
  }
}
