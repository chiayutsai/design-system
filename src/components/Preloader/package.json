{
  "name": "Preloader",
  "version": "0.0.0",
  "private": true,
  "main": "Preloader.js"
}
