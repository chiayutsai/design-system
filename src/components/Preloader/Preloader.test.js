import renderer from 'react-test-renderer'
import MockAppWrapper from 'components/MockAppWrapper'
import Preloader from './Preloader'

describe('Preloader', () => {
  test('render correctly', () => {
    const wrapper = renderer
      .create(
        <MockAppWrapper>
          <Preloader />
        </MockAppWrapper>,
      )
      .toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
