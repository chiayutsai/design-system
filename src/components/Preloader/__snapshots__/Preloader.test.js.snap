// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader render correctly 1`] = `
<div
  className="preloader"
>
  <div
    className="preloader_spiner"
  >
    <div
      className="preloader_circles"
    >
      <div
        className="preloader_circle-1"
      />
      <div
        className="preloader_circle-2"
      />
    </div>
  </div>
</div>
`;
